x-pack/plugins/security/server/routes/user_profile/get.ts

```diff
@@ -8,26 +8,42 @@
 import { schema } from '@kbn/config-schema';
 
 import type { RouteDefinitionParams } from '../';
+import type { AuthenticatedUserProfile } from '../../../common/';
 import { wrapIntoCustomErrorResponse } from '../../errors';
 import { createLicensedRouteHandler } from '../licensed_route_handler';
 
 export function defineGetUserProfileRoute({
   router,
+  getSession,
   getUserProfileService,
+  logger,
 }: RouteDefinitionParams) {
   router.get(
     {
-      path: '/internal/security/user_profile/{uid}',
-      options: { tags: ['access:accessUserProfile'] },
+      path: '/internal/security/user_profile',
       validate: {
-        params: schema.object({ uid: schema.string() }),
+        query: schema.object({ data: schema.maybe(schema.string()) }),
       },
     },
     createLicensedRouteHandler(async (context, request, response) => {
+      const session = await getSession().get(request);
+      if (!session) {
+        logger.warn('User profile requested without valid session.');
+        return response.unauthorized();
+      }
```

Comment:
Follow-on from https://github.com/elastic/kibana/pull/127624/files#r827822036 --

This is actually expected when the user is authenticated in a sessionless manner (e.g., using [HTTP authentication](https://www.elastic.co/guide/en/kibana/8.1/kibana-authentication.html#http-authentication)). I updated your PR description with steps to test HTTP authentication using a snippet for a small Node reverse proxy server.

This code is problematic because the client attempts to fetch the profile, receives a 401 response error, and redirects to the login page. Then the client is authenticated, so it redirects to Home again, and it's an endless loop.

I think it would be better to respond with a 404 error here. CC @azasypkin in case he has another suggestion.

Reply:
> I think it would be better to respond with a 404 error here. CC @azasypkin in case he has another suggestion.

Yeah, I agree, 404 sounds like a better option. I initially assumed that we'll hide all profile-related links and suppress all "automatically triggered" profile-related API requests from client if we detect that user is authenticated with `http` provider, but I'm not sure if it's feasible to do.

Reply:
> I initially assumed that we'll hide all profile-related links and suppress all "automatically triggered" profile-related API requests from client if we detect that user is authenticated with `http` provider, but I'm not sure if it's feasible to do.

Same, on both counts.